File: BDA_backend/src/main/java/com/example/bda/Services/ConsultasEspacialesService.java
package com.example.bda.Services;

import com.example.bda.DTO.*;
import com.example.bda.Repository.ConsultasEspacialesRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ConsultasEspacialesService {

    @Autowired
    private ConsultasEspacialesRepository consultasRepository;

    // ============ DENSIDAD REAL ============
    
    public List<DensidadRealDTO> obtenerDensidadReal() {
        return consultasRepository.obtenerDensidadReal();
    }

    public List<DensidadRealDTO> obtenerDensidadRealPorZona(Long idZona) {
        return consultasRepository.obtenerDensidadRealPorZona(idZona);
    }

    // ============ PROXIMIDAD (ESCUELAS) ============
    
    public List<EscuelaCercanaDTO> obtenerEscuelasCercanas() {
        return consultasRepository.obtenerEscuelasCercanas();
    }

    public List<EscuelaCercanaDTO> obtenerEscuelasCercanasPorProyecto(Long idProyecto) {
        return consultasRepository.obtenerEscuelasCercanasPorProyecto(idProyecto);
    }

    // ============ SUPERPOSICIÓN DE PROYECTOS ============
    
    public List<ProyectoSuperpuestoDTO> obtenerProyectosSuperpuestos() {
        return consultasRepository.obtenerProyectosSuperpuestos();
    }

    public List<ProyectoSuperpuestoDTO> obtenerProyectosSuperpuestosPorId(Long idProyecto) {
        return consultasRepository.obtenerProyectosSuperpuestosPorId(idProyecto);
    }

    // ============ COBERTURA DE SERVICIOS ============
    
    public List<CoberturaServicioDTO> obtenerCoberturaServicios() {
        return consultasRepository.obtenerCoberturaServicios();
    }

    public CoberturaServicioDTO obtenerCoberturaServiciosPorZona(Long idZona) {
        return consultasRepository.obtenerCoberturaServiciosPorZona(idZona);
    }
}

File: BDA_frontend/src/services/ConsultasEspacialesService.js
import api from './api'

// URLs específicas para consultas espaciales
const API_URL = '/api/consultas-espaciales'

class ConsultasEspacialesService {
  // ============================================================
  // 1. CÁLCULO DE DENSIDAD REAL
  // ============================================================
  
  obtenerDensidadReal() {
    return api.get(`${API_URL}/densidad-real`)
  }

  obtenerDensidadRealPorZona(idZona) {
    return api.get(`${API_URL}/densidad-real/${idZona}`)
  }

  // ============================================================
  // 2. ANÁLISIS DE PROXIMIDAD (ESCUELAS)
  // ============================================================
  
  obtenerEscuelasCercanas() {
    return api.get(`${API_URL}/escuelas-cercanas`)
  }

  obtenerEscuelasCercanasPorProyecto(idProyecto) {
    return api.get(`${API_URL}/escuelas-cercanas/proyecto/${idProyecto}`)
  }

  // ============================================================
  // 3. SUPERPOSICIÓN DE PROYECTOS
  // ============================================================
  
  obtenerProyectosSuperpuestos() {
    return api.get(`${API_URL}/proyectos-superpuestos`)
  }

  obtenerProyectosSuperpuestosPorId(idProyecto) {
    return api.get(`${API_URL}/proyectos-superpuestos/${idProyecto}`)
  }

  // ============================================================
  // 4. COBERTURA DE SERVICIOS (HOSPITALES)
  // ============================================================
  
  obtenerCoberturaServicios() {
    return api.get(`${API_URL}/cobertura-servicios`)
  }

  obtenerCoberturaServiciosPorZona(idZona) {
    return api.get(`${API_URL}/cobertura-servicios/${idZona}`)
  }
}

export default new ConsultasEspacialesService()
